x-pack/plugins/cloud_integrations/cloud_experiments/public/plugin.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from '@kbn/core/public';
+import LaunchDarkly, { type LDClient } from 'launchdarkly-js-client-sdk';
+import { get, has } from 'lodash';
+import type {
+  CloudExperimentsFeatureFlagNames,
+  CloudExperimentsMetric,
+  CloudExperimentsPluginSetup,
+  CloudExperimentsPluginStart,
+} from '../common';
+import { FEATURE_FLAG_NAMES, METRIC_NAMES } from '../common/constants';
+
+/**
+ * Browser-side implementation of the Cloud Experiments plugin
+ */
+export class CloudExperimentsPlugin
+  implements Plugin<CloudExperimentsPluginSetup, CloudExperimentsPluginStart>
+{
+  private launchDarklyClient?: LDClient;
+  private readonly clientId?: string;
+  private readonly kibanaVersion: string;
+  private readonly flagOverrides?: Record<string, unknown>;
+
+  /** Constructor of the plugin **/
+  constructor(initializerContext: PluginInitializerContext) {
+    this.kibanaVersion = initializerContext.env.packageInfo.version;
+    const config = initializerContext.config.get<{
+      launch_darkly?: { client_id: string };
+      flag_overrides?: Record<string, unknown>;
+    }>();
+    if (config.flag_overrides) {
+      this.flagOverrides = config.flag_overrides;
+    }
+    const ldConfig = config.launch_darkly;
+    if (!ldConfig && !initializerContext.env.mode.dev) {
+      // If the plugin is enabled, and it's in prod mode, launch_darkly must exist
+      // (config-schema should enforce it, but just in case).
+      throw new Error(
+        'xpack.cloud_integrations.experiments.launch_darkly configuration should exist'
+      );
+    }
+    if (ldConfig) {
+      this.clientId = ldConfig.client_id;
+    }
+  }
+
+  /**
+   * Returns the contract {@link CloudExperimentsPluginSetup}
+   * @param core {@link CoreSetup}
+   */
+  public setup(core: CoreSetup): CloudExperimentsPluginSetup {
+    return {
+      identifyUser: (userId, userMetadata) => {
+        if (!this.clientId) return; // Only applies in dev mode.
+
+        if (!this.launchDarklyClient) {
+          // If the client has not been initialized, create it with the user data..
+          this.launchDarklyClient = LaunchDarkly.initialize(
```

Comment:
I know we are making this plugin disable-able, but it would still be really nice if there were anything we can do to load the LD lib async instead of having it add to the weight of the initial page bundle for all cloud users. 

- We could make all of the methods (identifyUser, getVariation, reportMetric) async in order to do an async import of the SDK the first time one of the methods is called 
- We could put everything behind some sort of `await cloudExperiments.initialize()` method
- We could consider loading it async in the background (without `await`ing), and buffering calls to the methods that happen until the SDK is available, then replaying them back

None of those options are great, but just wanted to bring it up to see if you had any other ideas. It's probably something we can consider as a follow-up discussion.

Reply:
My main concern with making the identify call async is that it should run in the setup method. 

But I see the value in what you say. FWIW, the LD client that I used takes 45kB. But everything adds up, so the more we can lazy-load, the better.

In the future, we may need to retrieve the deployment metadata via async requests, so, probably, also making the identify call async wouldn't be that much of a problem.

Reply:
I've played a bit with it. FWIW, we are planning to revert the dependency with `cloud`, and that will remove the `identifyUser` API. So I think we can optimize it afterwards. This way we don't push the promise handling down to the `cloud` plugin in this version.

IMO, the `reportMetric` API should still be synchronous. Usually, `track` APIs are synchronous and push the queueing logic and async operations internally. I think we could do the same here and enqueue them internally if we think it's worth it in the future (or even leave it as it is right now: calling it before the client is initialized will simply skip reporting it).

What do you think?

Reply:
That all makes sense to me. +1 to waiting until we invert the `cloud` dependency to revisit, and to considering a queuing approach to keep `track` calls sync.